addons/crm/models/crm_lead.py

```diff
@@ -387,6 +391,38 @@ def _compute_name(self):
             if not lead.name and lead.partner_id and lead.partner_id.name:
                 lead.name = _("%s's opportunity") % lead.partner_id.name
 
+    @api.depends('partner_id', 'partner_name')
+    def _compute_commercial_partner_id(self):
+        partnered_leads = self.filtered('partner_id')
+        for lead in partnered_leads:
+            commercial_partner = lead.partner_id.commercial_partner_id
+            lead.commercial_partner_id = commercial_partner.is_company and commercial_partner
```

Comment:
Note that in case of a single company, isn't commercial_partner_id himself ? We should probably also skip it if commercial_partner == lead.partner_id

Reply:
Sure, I think either is fine. Don't believe it actually changes any flow actually so it's just a choice to make.